### PR TITLE
Honor the pretrained argument after reset_weights() and in MultiTrainer

### PR DESCRIPTION
**Pull request by raimbekovm**
## summary

- `reset_weights()` records `pretrained=False` and `train()` resolves `pretrained` from the merged args before the multi-dataset branch, so the trainer builds after `init_seeds` instead of taking the unseeded reset module as the donor
- `load(<path>)` no longer stores the path in `overrides["pretrained"]`: the DDP launcher ships the parent's module since #26050, and that stored path is why `train()` ignored the override
- `MultiTrainer` no longer forces `pretrained=True` or forwards `cfg=` to its CLI children, and resolves the donor like `Model.train()` in the Python arm
- the classify trainer skips its `pretrained=False` reset on resume, as the base trainer does
- one commit per fix

## measured

cpu, coco8, yolo26n; w0 is the absolute sum of `model.0.conv.weight` in `last.pt` (90.58 = `yolo26n.pt`, 41-42 = fresh init)

| sequence | before | after |
| -- | -- | -- |
| `YOLO(pt).reset_weights().train(seed=0, deterministic=True)` twice, also with `data=[...]` or `cfg=<yaml without pretrained>` | differ | identical |
| `YOLO(yaml).train(data=[...], pretrained=pt)` / `.tune(pretrained=pt)` | 41.0 / 42.8 | 90.58 / 90.58 |
| `YOLO(pt).train(data=[...], pretrained=False)` | 90.58 | fresh init |
| `YOLO(pt).train(cfg=<copy of default.yaml>, data=[...])` per-dataset metrics | `None` | metrics |
| `YOLO(cls.pt).train(pretrained=False)`, interrupted, `train(resume=True)`: weights equal `last.pt` | no | yes |
| `YOLO(yaml).train(seed=0)` twice; `YOLO(pt).train()`; `YOLO("yolo11-cls-resnet18.yaml").train(pretrained=False)` scratch; YOLOE prompt-embedding id | unchanged | unchanged |

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Training now consistently honors the resolved `pretrained` setting after weight resets, checkpoint loads, and multi-dataset setup.

### 📊 Key Changes
- `reset_weights()` records `pretrained=False`, while `Model.train()` resolves `pretrained` from the merged training arguments before entering the multi-dataset path.
- `load()` no longer stores a checkpoint path in `overrides["pretrained"]`, preventing that path from overriding explicit training arguments.
- `MultiTrainer` now removes the already-merged `cfg` before launching CLI children and selects donor weights according to `pretrained` in Python-based training.
- Classification training no longer resets model parameters when resuming with `pretrained=False`.
- YOLO prompt-embedding model identification now uses the model checkpoint path or YAML source instead of `overrides["pretrained"]`.

### 🎯 Purpose & Impact
- `reset_weights().train()` can build the model after seed initialization rather than reusing the unseeded reset module.
- Explicit `pretrained=False` starts from fresh parameters, while a checkpoint path can be used as the donor in standard and multi-dataset training.
- Classification resume runs preserve the resumed weights instead of resetting them.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @raimbekovm, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message confirms your contribution was received, and an Ultralytics engineer will assist with the review. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Reviewed all five changed files and the relevant model, trainer, DDP, classification-resume, tuning, and prompt-embedding call paths. One medium regression remains: reset_weights() is still ignored when train() receives a cfg that omits pretrained.

<details><summary>💬 Posted 1 inline comment</summary>

- 💡 **MEDIUM** `ultralytics/engine/model.py:849` When `cfg` is supplied, `overrides` is replaced with `YAML.load(cfg)`, so the `pretrained=False` recorded by `reset_weights()` in `self.overrides` is dropped unless the training config also contains that key. Thus `YOLO(pt).reset_weights().train(cfg=...)` falls back to `pretrained=True` and reuses the unseeded reset module as the donor, defeating the seeded rebuild this change is intended to provide. Preserve the model override as the fallback while allowing explicit `cfg` or keyword values to…
</details>

File: ultralytics/engine/model.py [hunk 1]
```diff
@@ -309,6 +309,7 @@ def reset_weights(self) -> Model:
                 m.reset_parameters()
         for p in self.model.parameters():
             p.requires_grad = True
+        self.overrides["pretrained"] = False  # train() builds the seeded model from the architecture
```
Comment (UltralyticsAssistant) [inline ultralytics/engine/model.py]: 💡 **MEDIUM**: For a YAML-created model, `self.ckpt` is empty, so `Model.train()` skips the explicit `get_model()` rebuild and `BaseTrainer.setup_model()` returns immediately because it already has a module. Therefore this `pretrained=False` marker does not rebuild after `init_seeds`; repeated `YOLO(yaml).reset_weights().train(seed=0)` runs can still start from different weights. Make the rebuild path run for `pretrained is False` even when no checkpoint was loaded.

File: ultralytics/engine/model.py [hunk 2]
```diff
@@ -846,7 +846,7 @@ def train(
             )
             self.metrics = self.trainer.train()
             return self.metrics
-        pretrained = kwargs.get("pretrained", overrides.get("pretrained", True) if kwargs.get("cfg") else True)
+        pretrained = args.get("pretrained", self.overrides.get("pretrained", True))
```
Comment (UltralyticsAssistant) [inline ultralytics/engine/model.py]: 💡 **MEDIUM**: When `cfg` omits `pretrained`, this fallback is computed only after the `MultiTrainer` early return and is not written back into `args`. In the DDP path, the child trainer receives those args and `get_cfg()` fills the default `pretrained=True`, so `reset_weights()` is ignored; the Python `MultiTrainer` arm likewise defaults back to `True`. Resolve the fallback into `args` before the multi-dataset branch, e.g. with `pretrained = args.setdefault("pretrained", self.overrides.get("pretrained", True))`.

**Suggested change:**
```suggestion
        pretrained = args.setdefault("pretrained", self.overrides.get("pretrained", True))
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review 2

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Since the previous review, the cfg fallback is still open in DDP and MultiTrainer because it is resolved too late and is not propagated into trainer args. I also found a new YAML-model reset path where the pretrained marker has no effect because the existing module bypasses model construction. These can cause nondeterministic or incorrect initialization; remaining load, classification-resume, and prompt-embedding changes look consistent.

<details><summary>💬 Posted 2 inline comments</summary>

- 💡 **MEDIUM** `ultralytics/engine/model.py:312` For a YAML-created model, `self.ckpt` is empty, so `Model.train()` skips the explicit `get_model()` rebuild and `BaseTrainer.setup_model()` returns immediately because it already has a module. Therefore this `pretrained=False` marker does not rebuild after `init_seeds`; repeated `YOLO(yaml).reset_weights().train(seed=0)` runs can still start from different weights. Make the rebuild path run for `pretrained is False` even when no checkpoint was loaded.
- 💡 **MEDIUM** `ultralytics/engine/model.py:849` When `cfg` omits `pretrained`, this fallback is computed only after the `MultiTrainer` early return and is not written back into `args`. In the DDP path, the child trainer receives those args and `get_cfg()` fills the default `pretrained=True`, so `reset_weights()` is ignored; the Python `MultiTrainer` arm likewise defaults back to `True`. Resolve the fallback into `args` before the multi-dataset branch, e.g. with `pretrained = args.setdefault("pretrained", self.overrides.get("pretrained", True…
</details>

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review 3

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Since review 2, the cfg fallback is resolved into `args` before the MultiTrainer branch, and the CLI/Python MultiTrainer paths now honor the resolved pretrained donor while avoiding cfg reapplication. The load, classification-resume, and prompt-embedding changes remain consistent. The previously suspected YAML reset path is rebuilt by BaseTrainer from the model config after seeding, so no concrete defects remain. LGTM.

**Review by glenn-jocher — APPROVED** [review]
LGTM at 3e756155c5fac192330c6caf030da33939394f89. Completed a cold full-diff review against AGENTS.md; independent Claude Code `claude-fable-5-1` review also returns LGTM at this exact head.

Real one-epoch training checks passed for seeded resets, cfg fallback, MultiTrainer Python/CLI, pretrained=False and path donors, and default donors. An interrupted classification run with pretrained=False resumed with exact saved weights. Removed the in-scope mocked test.

CI is still running; this approval is a code review, not a merge authorization.

**Comment by UltralyticsAssistant** [comment]
🎉 PR [#26084](https://github.com/ultralytics/ultralytics/pull/26084) has been merged—thank you @raimbekovm, with contributions from @glenn-jocher!

> “Alone we can do so little; together we can do so much.” — Helen Keller

This collaboration improves training consistency by honoring resolved `pretrained` settings across weight resets, checkpoint loads, multi-dataset training, and classification resumes.

**Comment by glenn-jocher** [comment]
Thank you for fixing pretrained handling after reset_weights() and in MultiTrainer. Merged after validating real training, checkpoint resume, and MultiTrainer paths, plus an independent review. We also removed the mock-based test to keep the repair focused on the existing owners.
